src/python_testing/TC_CGEN_2_2.py

```
@@ -0,0 +1,850 @@
+        trusted_roots_list = await self.read_single_attribute_check_success(
+            dev_ctrl=self.default_controller,
+            node_id=self.dut_node_id,
+            cluster=self.cluster_opcreds,
+            attribute=self.cluster_opcreds.Attributes.TrustedRootCertificates)
+        trusted_roots_list_size = len(trusted_roots_list)
+        logger.info(f'Step #20 - TH1 Original size of the trusted_root list: {trusted_roots_list_size}')
+
+        self.step(21)
+        # Commissioning stage numbers - we should find a better way to match these to the C++ code
+        # CommissioningDelegate.h
+        # TODO: https://github.com/project-chip/connectedhomeip/issues/36629
+        kFindOperationalForCommissioningComplete = 30
+        logger.info(
+            f'Step #21 - TH2 Commissioning stage SetTestCommissionerPrematureCompleteAfter enum: {kFindOperationalForCommissioningComplete}')
+
+        resp = TH2.SetTestCommissionerPrematureCompleteAfter(kFindOperationalForCommissioningComplete)
+        logger.info('Step #21 - TH2 Commissioning DOES NOT send the CommissioningComplete command')
+
+        # TH2.SetSkipCommissioningComplete(True)
+        logger.info('Step #21 - CommissioningComplete skipped to avoid premature completion.')
+
+        self.step(22)
+        logger.info("Step #22 - TH1 Waiting for PASE session to stabilize...")
+        resp = await TH2.FindOrEstablishPASESession(setupCode=setup_qr_code, nodeid=newNodeId)
+        logger.info('Step #22 - TH2 successfully establish PASE session completed')
+
+        try:
+            # Verify DUT cannot proceed because the session has not been fully commissioned, leading to a timeout error
+            cmd = Clusters.GeneralCommissioning.Commands.ArmFailSafe(expiryLengthSeconds=0)
+            resp = await self.send_single_cmd(
+                dev_ctrl=TH2,
+                node_id=newNodeId,
+                cmd=cmd)
+        except ChipStackError as e:
+            asserts.assert_in('Timeout',
```

Is this a test plan error? Test plan says it should respond with OK. Is this caused by the pase session getting evicted while you're using it?

---

Yes, the issue comes from an incorrect expectation in the test plan for Step # 22, as it is expecting an "OK" response. However, the commissioning process is intentionally incomplete at Step # 21, as the `CommissioningComplete` command is not sent.

As a result, the PASE session is invalidated because the DUT is not fully commissioned. This leads to the `timeout error` when attempting to send further commands, as the session is no longer valid.

This is the reason I implemented the `except` block to handle the timeout error, which is the expected behavior due to the incomplete commissioning process.

I will update the Step # 22 to ensure that the implementation aligns with the expected behavior in the test plan (Current  updates on Test Plan PR [TC_CGEN_2_2 Test Plan Updated - Test Procedure and Table (new format) #4956](https://github.com/CHIP-Specifications/chip-test-plans/pull/4956)) as Open.

---

step 22 isn't updated in that PR.

---

I'm worried this is too SDK specific - this is testing all devices against what the SDK does. Really, the verification that the failsafe was successfully disarmed happens below, so can you just change this to be except: pass with an explanation?